learning/tour-of-beam/terraform/cloud_functions/variables.tf

```diff
@@ -0,0 +1,46 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+variable "service_account_id" {
+  description = "The name of Service Account to run Cloud Function"
+}
+
+variable "project_id" {
+  description = "The GCP Project ID of function"
+}
+
+variable "region" {
+  description = "The GCP Region of function"
+}
+
+variable "source_archive_bucket" {
+  description = "The GCS bucket containing the zip archive which contains the function"
```

Comment:
See comment above about using a resource-prefix to prevent having to ask about different variables for global resources.  In this example, perhaps one could do the following, continuing from the previous comment's example.  Instead of a `source_archive_bucket` variable:

```
locals {
   source_archive_bucket = "${var.resource_name_prefix}-archive-${random_string.id.result}"
}
```

Reply:
Prefix concept and random string has been implemented to generate some variable values
All the variables' values that are not import for the manual input are generated automatically using prefix and random string of 4 symbols